lib/graphql/schema/field.rb

```diff
@@ -646,30 +646,50 @@ def to_ruby_args(obj, graphql_args, field_ctx)
         if graphql_args.any? || @extras.any?
           # Splat the GraphQL::Arguments to Ruby keyword arguments
           ruby_kwargs = graphql_args.to_kwargs
+          maybe_lazies = []
           # Apply any `prepare` methods. Not great code organization, can this go somewhere better?
           arguments.each do |name, arg_defn|
             ruby_kwargs_key = arg_defn.keyword
 
-            loads = arg_defn.loads
-            if ruby_kwargs.key?(ruby_kwargs_key) && loads && !arg_defn.from_resolver?
+            if ruby_kwargs.key?(ruby_kwargs_key)
+              loads = arg_defn.loads
               value = ruby_kwargs[ruby_kwargs_key]
-              ruby_kwargs[ruby_kwargs_key] = if arg_defn.type.list?
-                value.map { |val| load_application_object(arg_defn, loads, val, field_ctx.query.context) }
+              loaded_value = if loads && !arg_defn.from_resolver?
+                if arg_defn.type.list?
+                  loaded_values = value.map { |val| load_application_object(arg_defn, loads, val, field_ctx.query.context) }
+                  if loaded_values.any? { |v| field_ctx.schema.lazy?(v) }
+                    GraphQL::Execution::Lazy.all(loaded_values)
+                  else
+                    loaded_values
+                  end
+                else
+                  load_application_object(arg_defn, loads, value, field_ctx.query.context)
+                end
               else
-                load_application_object(arg_defn, loads, value, field_ctx.query.context)
+                value
               end
-            end
 
-            if ruby_kwargs.key?(ruby_kwargs_key) && arg_defn.prepare
-              ruby_kwargs[ruby_kwargs_key] = arg_defn.prepare_value(obj, ruby_kwargs[ruby_kwargs_key])
+              maybe_lazies << field_ctx.schema.after_lazy(loaded_value) do |loaded_value|
+                ruby_kwargs[ruby_kwargs_key] = if arg_defn.prepare
+                  arg_defn.prepare_value(obj, loaded_value)
+                else
+                  loaded_value
+                end
+              end
             end
           end
 
           @extras.each do |extra_arg|
             ruby_kwargs[extra_arg] = fetch_extra(extra_arg, field_ctx)
           end
 
-          ruby_kwargs
+          if maybe_lazies.any? { |l| field_ctx.schema.lazy?(l) }
+            GraphQL::Execution::Lazy.all(maybe_lazies).then do
+              ruby_kwargs
+            end
+          else
+            ruby_kwargs
+          end
```

Comment:
This logic is duplicated so many places: "if any items in `a` are lazies, wrap `a` in a lazy, otherwise return this value. Maybe 

```
schema.after_any_lazies(l) {
  v 
}
```

Or, make `after_lazy` support Array natively 🤔

Reply:
Native array handling in `after_lazy` broke some existing behavior.